include/kframework/foundry.md

```diff
@@ -511,6 +511,18 @@ Finally, the original sender of the transaction, `ACCTFROM` is changed to the ne
       [priority(40)]
 ```
 
+We define a new rule for the `#halt ~> #return _ _` production that will trigger the `#endPrank` rules if the prank was set only for a single call.
+
+```{.k .bytes}
+    rule <k> (. => #endPrank) ~> #halt ~> #return _RETSTART _RETWIDTH ... </k>
+         <prank>
+           <singleCall> true </singleCall>
+           ...
+         </prank>
+      [priority(40)]
+
```

Comment:
Sholud we also be checking that we're at the same calldepth as before? That way we're not stopping and ending the prank on some nested call?

Reply:
I missed that. Thanks for pointing it out!